src/directives/formly-field.js

```diff
@@ -77,9 +77,41 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
         return undefined
       }
       if (angular.isDefined(newVal)) {
-        $scope.model[$scope.options.key] = newVal
+        parseSet($scope.options.key, $scope.model, newVal)
+      }
+      return parseGet($scope.options.key, $scope.model)
+    }
+
+    function parseSet(key, model, newVal) {
+      // If either of these are null/undefined then just return undefined
+      if (!key || !model) {
+        return
+      }
+      // If we are working with a number then $parse wont work, default back to the old way for now
+      if (angular.isNumber(key)) {
+        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric
+        model[key] = newVal
+      } else {
+        const setter = $parse($scope.options.key).assign
+        if (setter) {
```

Comment:
Is this `if` statement necessary? I wouldn't expect it would be. The `assign` property should always be a part of `$parse($scope.options.key)`.